.circleci/config.yml

```diff
@@ -0,0 +1,105 @@
+# DOCKER_EMAIL   - login info for docker hub
+# DOCKER_USER
+# DOCKER_PASS
+# DOCKER_DEPLOY - set to true to enable pushing to Dockerhub
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # base image
+      - image: ubuntu:16.04
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Install essential packages
+          command: |
+            apt-get update
+            apt-get install -y ca-certificates curl build-essential make git
+            curl -L -o /tmp/dockerize.tgz https://github.com/jwilder/dockerize/releases/download/v0.6.1/dockerize-linux-amd64-v0.6.1.tar.gz
+            tar -xz -C /usr/bin -f /tmp/dockerize.tgz
+      - run:
+          name: Create version.json
+          command: |
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+      - store_artifacts:
+          path: version.json
+      # FIXME: should use an image w/ docker installed by default
+      - run:
+          name: Install Docker
+          command: |
+            set -x
+            VER="17.12.1-ce"
+            curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$VER.tgz
+            tar -xz -C /tmp -f /tmp/docker-$VER.tgz
+            mv /tmp/docker/* /usr/bin
+      - run:
+          # Create DB now so it has time to initialize while the container build and
+          # tests run
+          name: Create DB
+          command: >-
+            docker run --name mysql
+            -e MYSQL_ALLOW_EMPTY_PASSWORD=yes
+            -e MYSQL_DATABASE=tokenserver_test
+            -e MYSQL_USER=ubuntu
+            -e MYSQL_PASSWORD=ubuntu
+            -d mysql:5.7.22
+      - run:
+          name: Build deployment container image
+          command: docker build -t app:build .
+      - run:
+          name: Test flake8
+          command: docker run app:build /bin/sh -c "flake8 tokenserver"
+      - run:
+          name: Wait for DB
+          command: >-
+            docker run
+            --link mysql:mysql
+            jwilder/dockerize -wait tcp://mysql:3306 -timeout 120s
+      - run:
+          name: Test nose
+          command: >-
+            docker run
+            --link mysql:mysql
+            --env "MOZSVC_SQLURI=pymysql://ubuntu:ubuntu@mysql/tokenserver_test"
+            app:build /bin/sh -c "nosetests tokenserver/tests"
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # set DOCKER_DEPLOY=true in Circle UI to push to Dockerhub
+            DOCKER_DEPLOY="${DOCKER_DEPLOY:-false}"
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              bin/ci/deploy-dockerhub.sh latest
+            fi
+            if [[ "${CIRCLE_BRANCH}" == dockerpush* ]]; then
```

Comment:
For consistency with other repos, could we please add `[[ "${CIRCLE_BRANCH}" == feature* ]]` here as well?

Reply:
Done